runner/providers/external/external.go

```diff
@@ -74,6 +79,7 @@ func (e *external) CreateInstance(ctx context.Context, bootstrapParams commonPar
 		fmt.Sprintf("GARM_POOL_ID=%s", bootstrapParams.PoolID),
 		fmt.Sprintf("GARM_PROVIDER_CONFIG_FILE=%s", e.cfg.External.ConfigFile),
 	}
+	asEnv = append(asEnv, e.environmentVariables...)
```

Comment:
We should probably dedup the variables. We also need to make sure that the values in `asEnv` take precedence to those in the env. We don't want to accidentally override `GARM_COMMAND` for example.

I know we have an include list of allowed environment variables, so unless we explicitly set `GARM_` in the config to allow those variables to be passed through, there is no chance of collision, but it may not be a bad idea to account for mistakes.

Or we can skip any pattern set in the config that starts with `GARM`, ensuring that we can't accidentally override the variables set by GARM.

Reply:
sounds totally valid - thanks for the finding. Will skip `GARM_` prefixed variables